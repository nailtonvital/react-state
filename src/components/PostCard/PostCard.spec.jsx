import { render, screen } from "@testing-library/react"
import { PostCard } from "."

const props = {
    title: 'title',
    body: 'body 1',
    id: 1,
    cover: 'img/img.png'
}

describe('<PostCard />', ()=>{
    it('shoud render PostCard correctly', ()=>{
        render(<PostCard {...props}/>)

        expect(screen.getByAltText(/title/))
        .toHaveAttribute('src','img/img.png')
        expect(screen.getByRole('heading', { name: /title/i})).toBeInTheDocument()
        expect(screen.getByText('body 1')).toBeInTheDocument()
    })

    it('shoud match snapshot', () => {
        const { container } = render(<PostCard {...props} />)
        expect(container.firstChild).toMatchSnapshot()
    })
})