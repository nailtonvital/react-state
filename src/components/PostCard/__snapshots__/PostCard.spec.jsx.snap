// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> shoud match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      title
    </h1>
    <p>
      body 1
    </p>
  </div>
</div>
`;
